### PR TITLE
Forque sensor hardware

### PR DESCRIPTION
**Pull request by Tonkel**
A Hardware class for the Force/Torque sensor and a main script which launches a ROS controller manager and handles the sensor. Also, there is a launch file and a controller configuration file.

### CODE REVIEW

File: src/ForqueSensorHW.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        biasState = pr_hardware_interfaces::TRIGGER_IDLE;
+      }
+    } else {
+      std::string msg = "FT: " + std::to_string(data.wrench.force.x) + " " +
+                        std::to_string(data.wrench.force.y) + " " +
+                        std::to_string(data.wrench.force.z) + " " +
+                        std::to_string(data.wrench.torque.x) + " " +
+                        std::to_string(data.wrench.torque.y) + " " +
+                        std::to_string(data.wrench.torque.z);
+      // ROS_INFO(msg.c_str());
```
Comment (gilwoolee) [inline src/ForqueSensorHW.cpp]: Remove this

File: src/ForqueSensorHW.cpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    return false;
+  }
+  return true;
+}
+
+//==============================================================================
+void ForqueSensorHW::update() {
+
+  if (biasState == pr_hardware_interfaces::TRIGGER_REQUESTED) {
+    shouldCollectBiasData = true;
```
Comment (gilwoolee) [inline src/ForqueSensorHW.cpp]: I am confused why `bias` is collected when `TRIGGER` is requested. Could you explain?

File: src/main.cpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  ros::init(argc, argv, "forque_hardware_interface",
+            ros::init_options::NoSigintHandler);
+  ros::NodeHandle nh;
+  ros::CallbackQueue queue;
+  nh.setCallbackQueue(&queue);
+  signal(SIGINT, mySigintHandler);
+
+  // Starting other stuff
+  forqueSensorHW::ForqueSensorHW forqueSensorHW("forqueSensor",
+                                                "endEffectorFrameId", address);
```
Comment (gilwoolee) [inline src/main.cpp]: I wouldn't use `endEffectorFrameId` here. Also, does it make sense to use hardcoded `address`? Should we read these from command line args?

File: src/main.cpp [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+#include <controller_manager/controller_manager.h>
+#include <forque_sensor_hardware/ForqueSensorHW.h>
+#include <geometry_msgs/WrenchStamped.h>
+#include <iostream>
+#include <ros/callback_queue.h>
+#include <ros/ros.h>
+
+bool shouldShutdown = false;
+
+void mySigintHandler(int sig) { shouldShutdown = true; }
```
Comment (gilwoolee) [inline src/main.cpp]: Can we make it `sigintHandler`?

File: include/forque_sensor_hardware/ForqueSensorHW.h [hunk 5]
```diff
@@ -0,0 +1,64 @@
+#ifndef FORCE_TORQUE_SENSOR_HW_H_
+#define FORCE_TORQUE_SENSOR_HW_H_
+
+#include <iostream>
+#include <memory>
+#include <hardware_interface/force_torque_sensor_interface.h>
+#include <hardware_interface/robot_hw.h>
+#include <pr_hardware_interfaces/TriggerableInterface.h>
+#include <pr_hardware_interfaces/TriggerState.h>
+#include <netft_rdt_driver/netft_rdt_driver.h>
```
[the remark this answers is not in the document]
Comment (Tonkel) [reply include/forque_sensor_hardware/ForqueSensorHW.h]: Ah, yeah that wasn't clear. We use the net-ft-ros driver to communicate with the sensor.

File: src/ForqueSensorHW.cpp (hunk 2 above)
Comment (Tonkel) [reply src/ForqueSensorHW.cpp]: MoveUntilTouchController uses trigger to calibrate bias

**Pull request merged**